src/lib.rs

```diff
@@ -380,28 +418,43 @@ tuple_abomonate!(A B C D E F G H I J K L M N O P Q R S T U V W X Y Z AA AB AC AD
 tuple_abomonate!(A B C D E F G H I J K L M N O P Q R S T U V W X Y Z AA AB AC AD AE);
 tuple_abomonate!(A B C D E F G H I J K L M N O P Q R S T U V W X Y Z AA AB AC AD AE AF);
 
+impl<T: Abomonation> Abomonation for [T] {
+    #[inline(always)]
+    unsafe fn entomb<W: Write>(&self, write: &mut W) ->  IOResult<()> {
+        for element in self { element.entomb(write)?; }
+        Ok(())
+    }
+
+    #[inline(always)]
+    unsafe fn exhume<'a>(self_: NonNull<Self>, bytes: &'a mut[u8]) -> Option<&'a mut [u8]> {
+        // FIXME: This (briefly) constructs an &[T] to invalid data, which is UB.
+        //        I'm not sure if this can be fully resolved without relying on slice implementation details.
+        let self_len = self_.as_ref().len();
+        exhume_slice( self_.as_ptr() as *mut T, self_len, bytes )
+    }
+
+    #[inline(always)] fn extent(&self) -> usize {
+        self.iter().map(|inner| inner.extent()).sum()
+    }
+}
```

Comment:
Whoops, this experiment slipped in but is actually only weakly related to this PR. Tell me it you want it moved somewhere else.

Reply:
Note from the future of this PR: this is actually a rather bad idea, because one day or another someone will want to implement `Abomonation` for `&[T]` (_cough cough_), and having a trait implemented for both `[T]` and `&[T]` interacts with the `Deref` machinery in a dangerously surprising way.

I must find another way to factor out commonalities between slice-like types.

Reply:
...or maybe I can find a way to make it work. But that's very much worthy of a new Github issue.